### PR TITLE
fix: App locator create app flow

### PR DESCRIPTION
**Pull request by deepak-devtron**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/common/header/HeaderWithCreateButton/HeaderWithCreateButton.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ export default function HeaderWithCreateButton({ headerName }) {
     const renderCreateSelectionModal = () => {
         return (
             <Modal rootClassName="create-modal-wrapper" onClick={handleCreateButton}>
-                <div className="create-modal-child cursor" onClick={openCreateDevtronAppModel}>
+                <div className="create-modal-child cursor" onClick={openCreateDevtronAppModel} data-testid="create-custom-app-button-in-dropdown">
```
Comment (shivani170) [inline src/components/common/header/HeaderWithCreateButton/HeaderWithCreateButton.tsx:57]: please format also the lines which been updated

File: src/components/app/create/CreateApp.tsx [hunk 2]
```diff
@@ -286,6 +286,7 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
                     <span className="form__label dc__required-field">{this.props.isJobView ? 'Job' : 'App'} Name</span>
                     <input
                         ref={(node) => (this._inputAppName = node)}
+                        data-testid="app_name_textBox"
```
Comment (vivek-devtron) [inline src/components/app/create/CreateApp.tsx:289]: testid names should be in standard format either use snake_case or kebab-case

File: src/components/ciConfig/CIAdvancedConfig.tsx [hunk 3]
```diff
@@ -97,6 +97,7 @@ export default function CIAdvancedConfig({
                         <div className="flexbox justify-space" key={`build-${idx}`}>
                             <div className="mt-8 w-100">
                                 <input
+                                    data-testid="build-pack-build-env-key"
```
Comment (deepak-devtron) [inline src/components/ciConfig/CIAdvancedConfig.tsx:100]: why hardcoded , use index here

File: src/components/ciConfig/CIAdvancedConfig.tsx [hunk 4]
```diff
@@ -111,6 +112,7 @@ export default function CIAdvancedConfig({
                                     disabled={updateNotAllowed}
                                 />
                                 <textarea
+                                    data-testid="build-pack-build-env-value"
```
Comment (deepak-devtron) [inline src/components/ciConfig/CIAdvancedConfig.tsx:115]: use index here

File: src/components/charts/list/DiscoverCharts.tsx [hunk 5]
```diff
@@ -406,6 +407,7 @@ function DiscoverChartList() {
                                                                                       )
                                                                                     : selectChart(chartId)
                                                                             }
+                                                                            datatestid={`single-${index}`}
```
Comment (deepak-devtron) [inline src/components/charts/list/DiscoverCharts.tsx:410]: rename it , single-0, single-1 doesn't make any sense here @Shivam-nagar23

File: src/components/charts/MultiChartSummary.tsx [hunk 6]
```diff
@@ -104,6 +110,7 @@ const MultiChartSummary: React.FC<MultiChartSummaryProps> = ({
                                 getChartVersionsAndValues ? () => getChartVersionsAndValues(chart.id, index) : null
                             }
                             hideDeployedValues={hideDeployedValues}
+                            datatestid={`${index}`}
```
Comment (deepak-devtron) [inline src/components/charts/MultiChartSummary.tsx:113]: add some prefix as well before index

File: src/components/charts/ChartGroupUpdate.tsx [hunk 7]
```diff
@@ -327,6 +327,7 @@ function ChartList({ availableCharts, selectedInstances, addChart, subtractChart
                     subtractChart={subtractChart}
                     showCheckBoxOnHoverOnly={false}
                     showDescription={!isGrid}
+                    datatestid={`${idx}`}
```
Comment (deepak-devtron) [inline src/components/charts/ChartGroupUpdate.tsx:330]: add some prefix as well

File: src/components/app/details/cicdHistory/History.components.tsx [hunk 8]
```diff
@@ -81,9 +81,11 @@ export const GitChanges = ({ gitTriggers, ciMaterials }: GitChangesType) => {
                     <div
                         key={`mat-${gitTrigger?.Commit}-${index}`}
                         className="bcn-0 pt-12 br-4 en-2 bw-1 pb-12 mb-12"
+                        data-testid="source-code-git-hash"
                         style={{ width: 'min( 100%, 800px )' }}
                     >
                         <GitCommitInfoGeneric
+                            index={index}
```
Comment (deepak-devtron) [inline src/components/app/details/cicdHistory/History.components.tsx:88]: what is the requirement of this

File: src/components/CustomChart/CustomChartList.tsx [hunk 9]
```diff
@@ -153,7 +153,7 @@ export default function CustomChartList() {
             <div className="chart-list">
                 <div className="cn-9 fw-6 fs-16">Custom charts</div>
                 {renderSubtitleAndUploadButton('Manage custom charts to be used in Devtron applications.')}
-                <div className="mt-16 en-2 bw-1 bcn-0 br-8" style={{ minHeight: 'calc(100vh - 235px)' }}>
+                <div dataTest-id="custom-chart-list" className="mt-16 en-2 bw-1 bcn-0 br-8" style={{ minHeight: 'calc(100vh - 235px)' }}>
```
[the remark this answers is not in the document]
Comment (kushagra-devtron) [reply src/components/CustomChart/CustomChartList.tsx:156]: @rish2320

File: src/components/workflowEditor/NoGitOpsConfiguredWarning.tsx [hunk 10]
```diff
@@ -29,6 +29,7 @@ export default function NoGitOpsConfiguredWarning({ closePopup }: NoGitOpsConfig
                 </div>
                 <div className="flex right confirmation-dialog__button-group">
                     <button
+                        data-testid="continue_with-helm"
```
[the remark this answers is not in the document]
Comment (kushagra-devtron) [reply src/components/workflowEditor/NoGitOpsConfiguredWarning.tsx:32]: @dhananjay0601

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 11]
```diff
@@ -1014,7 +1014,7 @@ function ChartValuesView({
                     <Edit className="icon-dim-12 mr-6" />
                     {ConfigurationType.YAML}
                 </RadioGroup.Radio>
-                <RadioGroup.Radio value="manifest" canSelect={isValidData()} tippyContent={MANIFEST_INFO.InfoText}>
+                <RadioGroup.Radio value="manifest" canSelect={isValidData()} tippyContent={MANIFEST_INFO.InfoText} dataTestId="manifest-radio-button">
```
[the remark this answers is not in the document]
Comment (kushagra-devtron) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:1017]: @Shivam-nagar23

File: src/components/v2/common/message.ui.tsx [hunk 12]
```diff
@@ -31,6 +32,7 @@ export interface MsgUIProps {
 
 const MessageUI: React.FC<MsgUIProps> = ({
     msg,
+    dataTestId="",
```
[the remark this answers is not in the document]
Comment (kushagra-devtron) [reply src/components/v2/common/message.ui.tsx:35]: @Ash-exp

File: src/components/app/details/cicdHistory/History.components.tsx (hunk 8 above)
Comment (kushagra-devtron) [reply src/components/app/details/cicdHistory/History.components.tsx:88]: @Aravind-2023

Comment (kushagra-devtron) [reply src/components/app/details/cicdHistory/History.components.tsx:88]: used in GitCommitInfoGeneric.tsx

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**